pod.go

```diff
@@ -87,49 +90,27 @@ func (p *Pod) SetLabels(labels map[string]string) *Pod {
 
 // EmptyEnvironment empties the environment variables for a pod
 func (p *Pod) EmptyEnvironment() *Pod {
-	p.Environment = make(map[string]interface{})
+	p.Environment = make(map[string]string)
 	return p
 }
 
 // AddEnvironment adds an environment variable to a pod
 func (p *Pod) AddEnvironment(name, value string) *Pod {
+	if p.Environment == nil {
+		p = p.EmptyEnvironment()
+	}
 	p.Environment[name] = value
 	return p
 }
 
 // ExtendEnvironment extends the environment with the new environment variables
 func (p *Pod) ExtendEnvironment(env map[string]string) *Pod {
-	for k, v := range env {
-		p.AddEnvironment(k, v)
-	}
-	return p
-}
-
-// GetEnvironmentVariable gets the string contained in an environment variable
-func (p *Pod) GetEnvironmentVariable(name string) (string, error) {
-	str := ""
-	var err error
-
-	if val, ok := p.Environment[name]; ok {
-		switch val.(type) {
-		case string:
-			str = val.(string)
-			err = nil
-		case EnvironmentSecret:
-			err = fmt.Errorf("environment variable refers to a secret")
-		default:
-			err = fmt.Errorf("environment variable refers to unknown type")
-		}
-	} else {
-		err = fmt.Errorf("environment variable not found")
+	if p.Environment == nil {
+		p = p.EmptyEnvironment()
```

Comment:
We don't really need to empty the environment because `AddEnvironment` already takes care of that, right?